### ISSUE TITLE
[AST Builder] Implement Lower function

### ISSUE THREAD

**Issue by 24seconds** [gluesql/gluesql#614]
### Background
[`ast_builder`](https://github.com/gluesql/gluesql/blob/e8323b878851098ad6d2202427ba84b4b7d821ef/core/src/ast_builder/mod.rs) module is usually used to support `query builder`. For example, instead of typing hard coded string, we can write SQL statement using `ast_builder` module's functions. 

For example, we can build SQL statement using `table`, `select` and `build` function - which are functions of the `ast_builder` module.
```rs
#[test]
fn select() {
    let actual = table("App").select().build();
    let expected = "SELECT * FROM App";
    test(actual, expected);
}
```
[code pointer](https://github.com/gluesql/gluesql/blob/e8323b878851098ad6d2202427ba84b4b7d821ef/core/src/ast_builder/select/root.rs#L87-L107)

There are so many keywords to implement especially in the function part.
Currently, only `abs` function is implemented in the [`function.rs`](https://github.com/gluesql/gluesql/blob/e8323b878851098ad6d2202427ba84b4b7d821ef/core/src/ast_builder/expr/function.rs) and can be used as below.
https://github.com/gluesql/gluesql/blob/e8323b878851098ad6d2202427ba84b4b7d821ef/core/src/ast_builder/expr/function.rs#L38-L53

## Task
There are so many functions in the [`ast module`](https://github.com/gluesql/gluesql/blob/e8323b878851098ad6d2202427ba84b4b7d821ef/core/src/ast/function.rs). We need to implement all the corresponding functions in `ast_builder` module.  
This issue is about introducing `Lower` function in `ast_builder` module so that `query builder` can support `lower` function. After implementing it, we should pass the test below.

Test to pass
```rs
#[test]
fn function() {
    // Lower
    let actual = lower(text("ABC"));
    let expected = "LOWER('ABC')";
    test_expr(actual, expected);

    let actual = expr("HoHo").lower();
    let expected = "LOWER(HoHo)";
    test_expr(actual, expected);
}
```

Good luck!

**Note**: This issue is not for implementing how the `Lower` function should behave. It is for introducing new `query builder` feature about `Lower` function.

**Comment by 24seconds** [comment]
cc: @panarch

**Comment by yugeeklab** [comment]
Can I take this issue?

**Comment by panarch** [comment]
@yugeeklab I'd recommend you to make another one :D 
The first PR might be simple, so the important step all we need to experience is writing the new issue by ourselves.

This would be better for @24seconds to take, but he may leave this for a while for other contributors to refer.

**Comment by panarch** [comment]
resolved in #843
